### DISCUSSION TITLE
How to handle multiple schemas

### DISCUSSION THREAD

**Discussion by lszczygielek** [fluentmigrator/fluentmigrator#1509] [Q&A] [answered]
Hi,
I searched documentation and issues, but found no answer. How to handle multi-tenant scenario where each schema in database is for given tenant. From fluentmigrator perspective one database would have many schemas, each should have same tables and other stuff - how to instrument fluentmigrator for such case? How to work around one schema?
Regards.

**Comment by fubar-coder** [comment, marked as answer]
The only way to handle this is to run the migrator with different configurations. It contains a convention set that can be customized.

Here is what needs to be done step by step:

- Create a new service collection
- Add the FM services
- Add a scoped `IConventionSet` service
- Build the service provider
- Create a new service scope if needed (depends on your DI container)
- Instantiate the `IMigrationRunner`
- Call `MigrateUp` on the runner

Example:

```csharp
// Other configuration yadda-yadda ignored
services.AddScoped<IConventionSet>(_ => new DefaultConventionSet("your-custom-schema"));
using var serviceProvider = services.BuildServiceProvider();
using var scope = serviceProvider.CreateScope();
var runner = scope.ServiceProvider.GetRequiredService<IMigrationRunner>();
runner.MigrateUp();
```

The important part here is to provider your own custom convention set.

**Reply by lszczygielek** [reply]
Thank you for such full answer. It works great, it event create schema without additional migration.

**Reply by lszczygielek** [reply]
~~I have one more question. Am I right that such configuration (passing schema name) requires to write own application for db migration, because build-in tools doeasn't allow for schema change?~~

No question. I just found parameter `--default-schema-name `.
